highfidelity/hifi
# WL 21545: Fix up the doppleganger mirror script

This PR implements a potential fix for the doppleganger "scrunching up into a ball" issue.


#### Test plan:
This is a pure JavaScript change -- so the current Beta release should be fine for testing.
* Load [app-doppleganger-marketplace.js](https://raw.githubusercontent.com/humbletim/hifi/708698f/unpublishedScripts/marketplace/doppleganger-attachments/dist/app-doppleganger-marketplace.js) from this PR branch.
* Use the "Mirror Mode" button to test doppleganger use across a few different avatar models and attachment combos. 
  ![image](https://user-images.githubusercontent.com/94753/30625905-43e34f8c-9d94-11e7-894a-8074a1e3c984.png)
* Verify that none of the dopplegangers appear scrunched up.

#### Deployment:
* Once approved the existing [Mirror Mode](https://highfidelity.com/marketplace/items/d7d6f41b-6e3e-49a3-b195-7afa0d91c6d0) marketplace item will need to have its app-doppleganger-marketplace.js script replaced with the one from this PR.
* Users running the earlier version will need to stop via Running Scripts... and then re-install from the marketplace. 

*Note: The underlying issue might be that `MyAvatar.getJointTranslations()` now returns translations in meters (as opposed to centimeters) -- or maybe the opposite with how underlying Model/Rig code works behind `ModelOverlay`'s application of the `jointTranslations` property.*

// WL item https://worklist.net/21545
